hooks/post_push

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Use manifest-tool to create the manifest, given the experimental
+# "docker manifest" command isn't available yet on Docker Hub.
+
+TAG_ROOT=${DOCKER_TAG%-*}
+TARGET_ARCH=""
+case "$DOCKER_TAG" in
+    *amd64)
+        TARGET_ARCH="amd64"
+        ;;
+    *arm32v5)
+        TARGET_ARCH="arm/v5"
+        ;;
+    *arm32v6)
+        TARGET_ARCH="arm/v6"
+        ;;
+    *arm32v7)
+        TARGET_ARCH="arm/v7"
+        ;;
+    *arm64v8)
+        TARGET_ARCH="arm64/v8"
+        ;;
+    *i386)
+        TARGET_ARCH="i386"
+        ;;
+    *ppc64le)
+        TARGET_ARCH="ppc64le"
+        ;;
+    *s390x)
+        TARGET_ARCH="s390x"
+        ;;
+    *)
+        echo "Unknown target arch, skip manifest push"
+        exit 0
+        ;;
+esac
+
+# If the tag is essentially only the arch, we use latest
+if [[ "$TAG_ROOT" == "$TARGET_ARCH" ]]; then
+    TAG_ROOT="latest"
+fi
+
+curl -Lo manifest-tool https://github.com/estesp/manifest-tool/releases/download/v1.0.0/manifest-tool-linux-amd64
+chmod +x manifest-tool
```

Comment:
Did you consider using ``docker manifest``? I don't like the idea of using a random third party tool if we have a first party one which does the same job.

Reply:
I did do a bit of reading on it, but I'm not totally familiar with the `docker manifest` command.

It looks like what the tool is doing is definitely doable with `docker manifest` though not in the same way. Until I dig in I won't know if it's simpler, just as simple, or more complicated than the current method.

Now is as good a time as any for me to learn it though.

Reply:
Oh, right. There was another reason. The big warning on the top of the page of documentation:

> This command is experimental on the Docker client.
> 
> It should not be used in production environments.

https://docs.docker.com/engine/reference/commandline/manifest/

I actually put a comment at the top of this file explaining that, but I forgot.

There are also an experimental update to `docker build` that will allow specifying `--platform`. Once that and the manifest changes are GA, then this whole thing will become a lot easier. I wonder if at that point Docker Hub will allow proper support for multi-arch builds.

Reply:
I used it a look in the past but almost completely removed it after switching to buildx. I still use it here https://github.com/SuperSandro2000/docker-images/blob/master/archisteamfarm/manifest.sh#L20 .

I hope this helps you and if you have any question about it feel free to ask me.